packages/xo-server-sdn-controller/src/index.js

```diff
@@ -444,64 +443,19 @@ class SDNController extends EventEmitter {
           )
         }
       }
-      for (let i = 0; i < tunnels.length; ++i) {
-        const tunnel = tunnels[i]
-        const accessPIF = await xapi._getOrWaitObject(tunnel.access_PIF)
-        if (accessPIF.host !== host.$ref) {
-          continue
-        }
-
-        const poolNetwork = find(this._poolNetworks, {
-          network: accessPIF.network,
-        })
-        if (poolNetwork == null) {
-          continue
-        }
+    }
+  }
 
-        if (accessPIF.currently_attached) {
-          continue
-        }
+  async _hostMetricsUpdated(hostMetrics) {
+    const xapi = hostMetrics.$xapi
 
-        log.debug(
-          `Pluging PIF: '${accessPIF.device}' for host: '${
-            host.name_label
-          }' on network: '${accessPIF.$network.name_label}'`
-        )
-        try {
-          await xapi.call('PIF.plug', accessPIF.$ref)
-        } catch (error) {
-          log.error(
-            `XAPI error while pluging PIF: '${accessPIF.device}' on host: '${
-              host.name_label
-            }' for network: '${accessPIF.$network.name_label}'`
-          )
-        }
+    const hosts = filter(xapi.objects.all, { $type: 'host' })
+    const host = find(hosts, { metrics: hostMetrics.$ref })
```

Comment:
The complexity is `O(n + m)`, and this may run very often, be careful of the perf impact.

You could use some data structures to improve the situation.

Reply:
Done.